src/sql/src/plan/statement/ddl.rs

```diff
@@ -2086,9 +2090,125 @@ pub fn plan_create_materialized_view(
 
     let MaterializedViewOptionExtracted {
         assert_not_null,
+        refresh,
         seen: _,
     }: MaterializedViewOptionExtracted = stmt.with_options.try_into()?;
 
+    let refresh_schedule = {
+        let mut refresh_schedule = RefreshSchedule::empty();
+        let mut on_commits_seen = 0;
+        for refresh_option_value in refresh {
+            if !matches!(refresh_option_value, RefreshOptionValue::OnCommit) {
+                scx.require_feature_flag(&ENABLE_REFRESH_EVERY_MVS)?;
+            }
+            match refresh_option_value {
+                RefreshOptionValue::OnCommit => {
+                    on_commits_seen += 1;
+                }
+                RefreshOptionValue::AtCreation => {
+                    unreachable!("REFRESH AT CREATION should have been purified away")
+                }
+                RefreshOptionValue::At(RefreshAtOptionValue { mut time }) => {
+                    transform_ast::transform(scx, &mut time)?; // Desugar the expression
+                    let ecx = &ExprContext {
+                        qcx: &QueryContext::root(scx, QueryLifetime::OneShot),
+                        name: "REFRESH AT",
+                        scope: &Scope::empty(),
+                        relation_type: &RelationType::empty(),
+                        allow_aggregates: false,
+                        allow_subqueries: false,
+                        allow_parameters: false,
+                        allow_windows: false,
+                    };
+                    let hir = plan_expr(ecx, &time)?.cast_to(
+                        ecx,
+                        CastContext::Implicit,
+                        &ScalarType::MzTimestamp,
+                    )?;
+                    // (mz_now was purified away to a literal earlier)
+                    let timestamp = hir
+                        .into_literal_mz_timestamp().
+                        ok_or_else(|| sql_err!(
+                            "REFRESH AT argument must be an expression that can be simplified and/or cast \
+                            to a constant whose type is mz_timestamp (calling mz_now() is allowed)"
+                        ))?;
+                    refresh_schedule.ats.push(timestamp);
+                }
+                RefreshOptionValue::Every(RefreshEveryOptionValue {
+                    interval,
+                    aligned_to,
+                }) => {
+                    let interval = Interval::from_str(interval.as_str())?;
+                    if interval.as_microseconds() <= 0 {
+                        sql_bail!("REFRESH interval must be positive; got: {}", interval);
+                    }
+                    if interval.as_microseconds() > Interval::new(0, 27, 0).as_microseconds() {
```

Comment:
Oh, maybe the DST thing is not an issue because we say the schedule is always computed in UTC?

Reply:
Ok, so looking at this more deeply, I don't think we need to ban intervals longer than 30 days. An interval of length 90 days is totally fine as long as the user said `90 days` and not `3mo`. I think the condition we actually want to check here is that `interval.months == 0`. And then we can just call `Interval::duration()?` and we're good; as mentioned above I think it's fine to say all days have exactly 24 hours because we can just document that the schedule proceeds according to UTC regardless of whatever timezone you put in your `ALIGNED TO <timestamptz>` clause.

Reply:
Would we need to care about leap seconds? The interval docs don't mention leap seconds anywhere but if we care then we may want to double check that they're ignored.

Reply:
I think the problem doesn't occur with leap seconds, because those are handled by slight adjustments to the epoch millis timestamp rather than changing the mapping from human-readable times like 11:00 to epoch milliseconds, which is what a daylight saving time transition does.

Reply:
Ok, I will
- check `interval.months == 0`
- call `Interval::duration()`
- document that all days are considered 24 hours, and call it out that this might mean a +-1 hour shift across DST boundaries.

(Eventually, I'll implement cron schedules too anyhow, because it's also needed for complex schedules like not refreshing on weekends. And at that point, if somebody doesn't like the +-1 hour shift of REFRESH EVERY, then they can use cron schedules instead.)

Reply:
Done. (Except for the documentation part, which will be in a follow-up PR.)